wiki/PlaceholderExpansion.md

```diff
@@ -1,397 +1,289 @@
+#### Full Example
+Please see the [Common parts](#common-parts) section for info on the other methods.
 
 ```java
 package at.helpch.placeholderapi.example.expansions;
 
 import org.bukkit.OfflinePlayer;
 import me.clip.placeholderapi.expansion.PlaceholderExpansion;
-import at.helpch.placeholderapi.example.SomePlugin;
 
-/**
- * This class will automatically register as a placeholder expansion 
- * when a jar including this class is added to the directory 
- * {@code /plugins/PlaceholderAPI/expansions} on your server.
- * <br>
- * <br>If you create such a class inside your own plugin, you have to
- * register it manually in your plugins {@code onEnable()} by using 
- * {@code new YourExpansionClass().register();}
- */
 public class SomeExpansion extends PlaceholderExpansion {
 
-    // We get an instance of the plugin later.
-    private SomePlugin plugin;
+    @Override
+    public String getAuthor(){
+        return "someauthor";
+    }
+    
+    @Override
+    public String getIdentifier(){
+        return "example";
+    }
 
-    /**
-     * Since this expansion requires api access to the plugin "SomePlugin" 
-     * we must check if said plugin is on the server or not.
-     *
-     * @return true or false depending on if the required plugin is installed.
-     */
     @Override
-    public boolean canRegister(){
-        return (plugin = (SomePlugin) Bukkit.getPluginManager().getPlugin(getRequiredPlugin())) != null;
+    public String getVersion(){
+        return "1.0.0";
+    }
+    
+    @Override
+    public String onRequest(OfflinePlayer player, String params){
+        if(params.equalsIgnoreCase("name")){
+            return player == null ? null : player.getName(); // "name" requires the player to be valid
+    
+        if(params.equalsIgnoreCase("placeholder1")){
+            return "Placeholder Text 1";
+    
+        if(params.equalsIgnoreCase("placeholder2")){
+            return "Placeholder Text 2";
+        
+        return null; // Placeholder is unknown by the Expansion
     }
+}
+```
 
-    /**
-     * The name of the person who created this expansion should go here.
-     * 
-     * @return The name of the author as a String.
-     */
+----
+## With a Plugin (External Jar)
+If your Expansion relies on a Plugin to provide its placeholder values you will need to override a few more methods to make sure everything will work as it should.
+
+Your expansion will need to override the `getRequiredPlugin()` method to return the name of the plugin your expansion depends on.  
+PlaceholderAPI does by default check, if this method either returns null, or the name defined results in a Not-null plugin being available.
+
+Something worth noting is, that it is a bit more difficult to make a separate Jar file that depends on a plugin, as it will require the plugin to have some sort of accessible API to use in order to get the values needed.  
```

Comment:
```suggestion
It is worth noting that it is a bit more difficult to make a separate jar file that depends on a plugin, as it will require the plugin to have some sort of accessible API in order to get the required values.
```